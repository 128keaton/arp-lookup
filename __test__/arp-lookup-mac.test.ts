jest.mock('child_process')

import arp from '../src/index'

describe('On a Mac environment', () => {
  Object.defineProperty(process, 'platform', {
    value: 'darwin',
  })
  test('Arp table is parsed', async () => {
    await expect(arp.getTable()).resolves.toMatchSnapshot()
  })
})
