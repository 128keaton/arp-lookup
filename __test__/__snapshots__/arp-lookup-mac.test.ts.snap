// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On a Mac environment Arp table is parsed 1`] = `
Array [
  Object {
    "ip": "192.168.137.255",
    "mac": "ff:ff:ff:ff:ff:ff",
    "type": "unknown",
  },
  Object {
    "ip": "224.0.0.22",
    "mac": "01:00:5e:00:00:16",
    "type": "unknown",
  },
  Object {
    "ip": "224.0.0.251",
    "mac": "01:00:5e:00:00:fb",
    "type": "unknown",
  },
  Object {
    "ip": "224.0.0.252",
    "mac": "01:00:5e:00:00:fc",
    "type": "unknown",
  },
  Object {
    "ip": "239.255.255.250",
    "mac": "01:00:5e:7f:ff:fa",
    "type": "unknown",
  },
  Object {
    "ip": "255.255.255.255",
    "mac": "ff:ff:ff:ff:ff:ff",
    "type": "unknown",
  },
  Object {
    "ip": "192.168.2.1",
    "mac": "04:a1:51:1b:12:92",
    "type": "unknown",
  },
  Object {
    "ip": "192.168.2.255",
    "mac": "ff:ff:ff:ff:ff:ff",
    "type": "unknown",
  },
  Object {
    "ip": "224.0.0.2",
    "mac": "01:00:5e:00:00:02",
    "type": "unknown",
  },
  Object {
    "ip": "224.0.0.22",
    "mac": "01:00:5e:00:00:16",
    "type": "unknown",
  },
  Object {
    "ip": "224.0.0.251",
    "mac": "01:00:5e:00:00:fb",
    "type": "unknown",
  },
  Object {
    "ip": "224.0.0.252",
    "mac": "01:00:5e:00:00:fc",
    "type": "unknown",
  },
  Object {
    "ip": "239.255.255.250",
    "mac": "01:00:5e:7f:ff:fa",
    "type": "unknown",
  },
  Object {
    "ip": "255.255.255.255",
    "mac": "ff:ff:ff:ff:ff:ff",
    "type": "unknown",
  },
]
`;
